prime-router/src/main/kotlin/azure/CheckFunction.kt

```diff
@@ -118,6 +122,78 @@ class CheckFunction : Logging {
         return HttpUtilities.httpResponse(request, responseBody.joinToString("\n") + "\n", httpStatus)
     }
 
+    /** @param code: "success" or "fail" enum for result of check's json response.
+     * want to limit to just one of the other */
+    enum class CheckResultsEnum(val code: String) { success("success"), fail("fail") }
+
+    /**
+     * Checks sftp settings works for a receiver.
+     *
+     * Differs from CheckFunction's `check()`:
+     *  - Uses OKTA oauth.
```

Comment:
Seems ok to just support okta, but I'd like our default to be support for both Okta and server2server.  There's a bunch of examples of this, eg, the Submission history queries.   You wouldn't need a new "scope" - just any primeadmin access works.

Reply:
More on this:     Maybe this support both Okta and server2server already?  AuthenticatedClaims.authenticate is supposed to support both token styles.

It always converts okta "group" claims into server2server-style "scope" claims, eg, *.*.primeadmin, which you are already using. 

So I suspectulate that this already works with server2server.   But not tested, I presume.

Reply:
I'm not sure how to test server2server. To date, I've only added APIs that use Okta OAuth. I'll go look at the submission history.

Reply:
Also, does this look correct?
```kotlin
        if (claims == null || !claims.authorized(setOf("*.*.primeadmin", "$orgName.*.admin"))) {
            logger.warn("User '${claims?.userName}' FAILED authorized for endpoint ${request.uri}")
            return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
        }

```